File: JavaScript/InsertionSort.js
"use strict"

function insertion_sort(array) {
    for (let i = 1; i < array.length; i++)
    {
        let j = i-1;
        while (j >= 0 && array[j] > array[j+1]) {
            [array[j], array[j + 1]] = [array[j + 1], array[j]];
            j--;
        }
    }
}

const ARRAY_LENGTH = 32;
const MAX_VALUE = 100;
let array = new Array(ARRAY_LENGTH);
for (let i = 0; i < array.length; i++) {
    array[i] = Math.floor(Math.random() * MAX_VALUE + 1);
}

console.log(array.join());
insertion_sort(array);
console.log(array.join());

File: C#/InsertionSort.cs

static void insertion_sort(int[] array, int array_length)
{
    for (int i = 1; i < array_length; i++)
    {
        int j = i-1;
        while (j >= 0 && array[j] > array[j+1]) {
            int aux = array[j];
            array[j] = array[j+1];
            array[j+1] = aux;

            j--;
        }
    }
}

const int ARRAY_LENGTH = 32;
const int MAX_VALUE = 100;
int[] array = new int[ARRAY_LENGTH];
Random random = new Random();
for (int i = 0; i < ARRAY_LENGTH; i++)
{
    array[i] = random.Next(MAX_VALUE + 1);
}

Console.WriteLine(string.Join(", ", array));
insertion_sort(array, ARRAY_LENGTH);
Console.WriteLine(string.Join(", ", array));

